tests/lib/modem_info/src/main.c

```diff
@@ -525,6 +525,118 @@ void test_modem_info_get_rsrp_success(void)
 	TEST_ASSERT_EQUAL(1, nrf_modem_at_scanf_fake.call_count);
 }
 
+void test_modem_info_RSRP_IDX_TO_DBM(void)
+{
+	int ret;
+
+	/* index < 0: index – 140 */
+	ret = RSRP_IDX_TO_DBM(-16);
+	TEST_ASSERT_EQUAL(-156, ret);
+
+	ret = RSRP_IDX_TO_DBM(-1);
+	TEST_ASSERT_EQUAL(-141, ret);
+
+	/* index > 0: index – 141 */
+	ret = RSRP_IDX_TO_DBM(1);
+	TEST_ASSERT_EQUAL(-140, ret);
+
+	ret = RSRP_IDX_TO_DBM(2);
+	TEST_ASSERT_EQUAL(-139, ret);
+
+	ret = RSRP_IDX_TO_DBM(96);
+	TEST_ASSERT_EQUAL(-45, ret);
+
+	ret = RSRP_IDX_TO_DBM(97);
+	TEST_ASSERT_EQUAL(-44, ret);
+
+	/* 0 defined as not used but we'll test because macro accepts anything */
+	ret = RSRP_IDX_TO_DBM(0);
+	TEST_ASSERT_EQUAL(-141, ret);
+
+	/* Values outside the range of AT command examples */
+	ret = RSRP_IDX_TO_DBM(-17);
+	TEST_ASSERT_EQUAL(-157, ret);
+
+	ret = RSRP_IDX_TO_DBM(98);
+	TEST_ASSERT_EQUAL(-43, ret);
+}
+
+void test_modem_info_RSRQ_IDX_TO_DB(void)
+{
+	int ret;
+
+	/* index < 0: (index – 39) / 2 */
+	ret = RSRQ_IDX_TO_DB(-30);
+	TEST_ASSERT_EQUAL(-34.5, ret);
```

Comment:
I guess this works because -34.5 gets implicitly converted into an integer. Still, looks a bit funny. Because both macros return a float, it would make sense to test these using floats.

Reply:
Good catch! I also added @param and @return statements into the header. Let's see from generated documentation if they look ok. If not, I'll remove the tags and include the same information in the text.